### ISSUE TITLE
Audit and Update Sketch Pattern Library

### ISSUE THREAD

**Issue by andaleliz** [department-of-veterans-affairs/vets-design-system-documentation#470]
## This update is for:

- [ ] Content styleguide
- [ ] Component
- [ ] Pattern
- [ ] Utility
- [x] Other: Sketch Pattern library, Storybook, Design.va.gov

## What is the nature of this update?

There are discrepancies between our Sketch pattern library, Storybook, and design.va.gov. Let's align these resources so VFS teams can be clear about which patterns and components they can use in their products without additional review and approval cycles. 

## Known discrepancies
- Design.va.gov suggests using an icon on buttons in some cases (second to last bullet [in this section](https://design.va.gov/components/buttons#how-to-use-buttons)), but there aren't any examples of this in the documentation (including storybook) or in Sketch. It'd be helpful to have examples.
- Sketch library contains background only alert symbols that have padding that is different than what is in Storybook and in the web component itself (20px on all sides in sketch vs 32px on top and 24px on sides).  The increased padding feels unnecessary, especially on mobile.
- [Storybook contains dismissible background-only alerts](https://design.va.gov/storybook/?path=/docs/components-va-alert--background-only). These are not clearly documented on design.va.gov, nor are they in our sketch pattern library.
- Sketch does not include the [sign-in or tool prompt alert](https://design.va.gov/components/alertboxes#sign-in-or-tool-prompt)
- Sketch does include a small success alert, but design.va.gov doesn't reference it

**Comment by andaleliz** [comment]
Additional context for this ticket: https://dsva.slack.com/archives/C0NGDDXME/p1621628112021900

**Comment by andaleliz** [comment]
FYSA @GnatalieH

**Comment by GnatalieH** [comment]
@andaleliz let's review the component file together in our meeting next week. Auditing the file is on my list of things to do. Thanks!

**Comment by GnatalieH** [comment]
Here is a breakdown of the tasks I've identified with this work:

1. Inventory of all symbols in the Sketch file
2. Archive deprecated components and patterns
3. Note where Sketch symbols and Storybook components don’t align
4. Perform cardsort/tree test with Designers to uncover how they would organize symbols intuitively
5. Create reorganized Sketch library and pilot with a few Designers 
6. Iterate as necessary

Sean (Product Support) is in the process of getting approval to conduct research for this project if we are cool with him helping, he has a lot of experience with design systems.

**Comment by caw310** [comment]
@GnatalieH , please write up tickets breaking up the tasks for this to be 3 or less so we can plan work for future sprint.

**Comment by caw310** [comment]
Adding two things Ryan mentioned that need updating:

Sketch library updates
- Forms elements has indent built into them. Left alignment is thrown off. 
- Headers and footers are out of date. Don’t render the way they do in real life.

**Comment by andaleliz** [comment]
Another thing - the card components in the Sketch file aren't in storybook or on design.va.gov. What is the status of using these cards, @caw310, @KevinMHoffmanUSDS  or @GnatalieH ? Are these available for teams to use? Here's one example my team is exploring:
<img width="452" alt="image" src="https://user-images.githubusercontent.com/55992116/161081878-faf45d5a-499f-4f5d-a791-74fb39abcdab.png">

**Comment by caw310** [comment]
Work for this request has been split up into Audit and Update. Tickets 674 and 675. We will update this ticket once both work on both those tickets is complete.

**Comment by GnatalieH** [comment]
Looping in @humancompanion-usds, the new Design System Team Product Owner. @andaleliz, we're aware the Cards in the pattern library file are not necessarily up to date. I believe we're going to delete most of the Cards in the process of this audit/update.  In the short term, I would suggest the following format for your card:

Title
Description
CTA below the Description 

Use 8px units of vertical spacing between Title, Description, and CTA. I'd probably put more space between the Description and the CTA than you put between the Title and Description. Also, if the CTA only takes the user to another page, use an Action link or link. If the CTA causes a state change, or "does something", a Secondary button (like the one in the card) is appropriate. 

That's my advice in the short-term while we finish this audit. Thanks!

**Comment by andaleliz** [comment]
Hi! This might already be on your radar, but I came across something else. In the sketch library the `ul` type style produces a bulleted list item with a round bullet and something like 20 px of padding between the bullet and the text starting.  Check out this screenshot:
![image](https://user-images.githubusercontent.com/55992116/166842261-671de5a2-8b52-422c-8737-56e680115d33.png)

On VA.gov, `ul` items have a square bullet and the text is much closer to the bullet.  Here's a screenshot of that from the [claim status page](https://www.va.gov/claim-or-appeal-status/)

<img width="465" alt="image" src="https://user-images.githubusercontent.com/55992116/166842294-64192fb2-b4fa-450f-983c-312b57df92ac.png">


I'm not sure which is "correct" but it'd be great if the Sketch file and CSS for the site were in alignment.

**Comment by humancompanion-usds** [comment]
@andaleliz - A couple of things here:

First, unordered lists: This is an artifact of how Sketch treats lists. Unfortunately, I can't control the spacing between the bullet and the text after the bullet. Nor in fact can we even define the bullet style in a Text Style. I tired defining a symbol but it's not going to address that spacing issue. 

One can change the bullet style at least to a square:
Look for "How to create and customise lists" here https://www.sketch.com/docs/designing/text/#text-options 

We could remove the Text Style entirely since it's not accurate. Or we could add a note in the bulleted items themselves to warn folks. Another option is to create a List symbol and keep the bullets and text separate. As long as your bulleted text is 1 line each then you can line them up manually or with line-height. It's ugly but it would be the most accurate match to production. Sketch's list styling leaves very much to be desired.

However, my feeling is that it doesn't really have a huge impact. So long as the bulleted list communicates a bulleted list both to the user in any research session and to the developer building it, then it's doing the job and will be correct once deployed. I do agree it is annoying and I'm not sure why Sketch doesn't let you at least define the text-options in your Text Style. 

Let me know what your preference is over Slack if you have strong feelings one way or the other. Otherwise I'll consult with Natalie and Sean on this and make a decision then take care of any changes. 

Secondly, I'm going to close this issue because we have completed the audit and queued up the work to update the Sketch library which is being tracked more actively here: 
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/675

We're also generating issues for updates as we work through that list. You had mentioned Cards which are tracked in this issue and which I plan to address and deploy this upcoming sprint (starts Thursday):
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/600

As for the items mentioned in this issue:
* Design.va.gov suggests using an icon on buttons in some cases (second to last bullet [in this section](https://design.va.gov/components/buttons#how-to-use-buttons)), but there aren't any examples of this in the documentation (including storybook) or in Sketch. It'd be helpful to have examples.
  * I've added this to the Button work we're doing this sprint: https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/va.gov-team/39874
 
* Sketch library contains background only alert symbols that have padding that is different than what is in Storybook and in the web component itself (20px on all sides in sketch vs 32px on top and 24px on sides). The increased padding feels unnecessary, especially on mobile.
  * This was done in our update to Alert in Sketch https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/va.gov-team/39872
 
* [Storybook contains dismissible background-only alerts](https://design.va.gov/storybook/?path=/docs/components-va-alert--background-only). These are not clearly documented on design.va.gov, nor are they in our sketch pattern library.
Sketch does not include the [sign-in or tool prompt alert](https://design.va.gov/components/alertboxes#sign-in-or-tool-prompt)
  * We added a dismissible alert style and a note about all alert variations being dismissible: https://design.va.gov/components/alert#dismissible
  * We've removed the Storybook dismissible alert because we can't show every option that can be applied in Storybook to a component in a distinct section or we'd have a very large page. 
  * We do now show all of the background-color only alert styles in both Storybook and design.va.gov.
  * If you feel there really needs to be a background alert shown as well as dismissible because that attribute is particularly important then go ahead an open a new issue for that distinctly.
  
* Sketch does include a small success alert, but design.va.gov doesn't reference it. 
  * Small success alert was removed from Sketch.
